packages/core/core/src/public/Bundle.js

```diff
@@ -66,123 +70,78 @@ export class Bundle implements IBundle {
     return this.#bundle.stats;
   }
 
-  getDependencies(asset: IAsset): Array<Dependency> {
-    return this.#bundle.assetGraph.getDependencies(
-      getInternalAsset(this.#bundle.assetGraph, asset)
+  hasAsset(asset: IAsset): boolean {
+    return this.#bundleGraph.bundleHasAsset(
+      this.#bundle,
+      assetToInternalAsset(asset)
     );
   }
 
+  getDependencies(asset: IAsset): Array<Dependency> {
+    return this.#bundleGraph.getDependencies(assetToInternalAsset(asset));
+  }
+
   getDependencyResolution(dependency: Dependency): ?Asset {
```

Comment:
In a follow-up, I'd like to get rid of this method and `getDependencies` on Bundle. Technically they can return assets/dependencies from outside the bundle, and conceptually I don't think these really belong.